### ISSUE TITLE
compile arror

### ISSUE THREAD

**Issue by daktras420** [iSparsh/gross#4]
from what i understand the maintainer of the aur build of gross is not the same maintainer as the original author so i am not sure whom to direct this issue too..but when compiling i get the following....

paru -S gross
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

Aur (1)        Old Version  New Version    Make Only
aur/gross-git               r28.b8d2b25-1  No

:: Proceed with installation? [Y/n]: y

:: Downloading PKGBUILDs...
 PKGBUILDs up to date
==> Making package: gross-git r28.b8d2b25-1 (Wed 16 Aug 2023 09:17:20 AM PDT)
==> Retrieving sources...
  -> Updating gross git repo...
==> Validating source files with md5sums...
    gross ... Skipped
==> Making package: gross-git r28.b8d2b25-1 (Wed 16 Aug 2023 09:17:21 AM PDT)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating gross git repo...
==> Validating source files with md5sums...
    gross ... Skipped
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Creating working copy of gross git repo...
Cloning into 'gross'...
done.
==> Starting prepare()...
==> Starting pkgver()...
==> Sources are ready.
gross-git-r28.b8d2b25-1: parsing pkg list...
==> Making package: gross-git r28.b8d2b25-1 (Wed 16 Aug 2023 09:17:23 AM PDT)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting pkgver()...
==> Removing existing $pkgdir/ directory...
==> Starting build()...
   Compiling autocfg v1.1.0
   Compiling libc v0.2.139
   Compiling proc-macro2 v1.0.56
   Compiling quote v1.0.26
   Compiling unicode-ident v1.0.6
   Compiling cfg-if v1.0.0
   Compiling pkg-config v0.3.27
   Compiling once_cell v1.16.0
   Compiling version_check v0.9.4
   Compiling syn v1.0.107
   Compiling bitflags v1.3.2
   Compiling futures-core v0.3.28
   Compiling memchr v2.5.0
   Compiling num-traits v0.2.15
   Compiling crossbeam-utils v0.8.15
   Compiling scopeguard v1.1.0
   Compiling memoffset v0.8.0
error[E0635]: unknown feature `proc_macro_span_shrink`
  --> /home/daniel/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.56/src/lib.rs:92:30
   |
92 |     feature(proc_macro_span, proc_macro_span_shrink)
   |                              ^^^^^^^^^^^^^^^^^^^^^^

   Compiling cc v1.0.78
   Compiling pin-project-lite v0.2.9
   Compiling crossbeam-epoch v0.9.14
For more information about this error, try `rustc --explain E0635`.
error: could not compile `proc-macro2` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'gross-git-r28.b8d2b25-1': 
error: packages failed to build: gross-git-r28.b8d2b25-1

when i run rustc --explain E0635 i get the following
The `#![feature]` attribute specified an unknown feature.

Erroneous code example:

```
#![feature(nonexistent_rust_feature)] // error: unknown feature



not sure how to fix this please help thanks in advance

**Comment by fabianmuehlberger** [comment]
Have you tried compiling with rustup ? 
I installed the aur package just a few days ago (plain arch linux) and changed to rustup due to deps of a package. Not sure if it helps tho.

**Comment by daktras420** [comment]
Yes i am using rustup and went through the whole arch install process to setup rustup and successful installed eww using the package from aur eww-tray-wayland-git. Despite all this as you can see it's not compiling.
